miri-script/src/commands.rs

```diff
@@ -0,0 +1,711 @@
+        if !extra_vars.libdir.exists() {
+            println!("Something went wrong determining the library dir.");
+            println!("I got {} but that does not exist.", extra_vars.libdir.display());
+            println!("Please report a bug at https://github.com/rust-lang/miri/issues.");
+            std::process::exit(2);
+        }
+        ret.extra_vars = extra_vars;
+        // Share target dir between `miri` and `cargo-miri`.
+        let target_dir = std::env::var("CARGO_TARGET_DIR")
+            .ok()
+            .filter(|val| !val.is_empty())
+            .unwrap_or_else(|| {
+                let target_dir = path!(ret.miri_dir / "target");
+                target_dir.to_string_lossy().to_string()
+            });
+        ret.extra_flags.insert("CARGO_TARGET_DIR".to_owned(), target_dir);
+
+        // We configure dev builds to not be unusably slow.
+        let devel_opt_level = std::env::var("CARGO_PROFILE_DEV_OPT_LEVEL")
+            .ok()
+            .filter(|val| !val.is_empty())
+            .unwrap_or_else(|| "2".to_owned());
+        ret.extra_flags.insert("CARGO_PROFILE_DEV_OPT_LEVEL".to_owned(), devel_opt_level);
+
+        let rustflags = std::env::var("RUSTFLAGS").unwrap_or_default();
+        let rustflags = format!(
+            "-Zunstable-options -Wrustc::internal -Wrust_2018_idioms -Wunused_lifetimes -Wsemicolon_in_expressions_from_macros {rustflags}"
+        );
+        let rustflags =
+            format!("-C link-args=-Wl,-rpath,{} {rustflags}", ret.extra_vars.libdir.display());
+        ret.extra_flags.insert("RUSTFLAGS".to_owned(), rustflags);
+        ret.execute()
+    }
+    fn execute(&mut self) -> Result<()> {
+        match self.command {
+            Subcommands::Install { flags } => self.install(&flags),
+            Subcommands::Build { flags } => self.build(&flags),
+            Subcommands::Check { flags } => self.check(&flags),
+            Subcommands::Test { flags } | Subcommands::Bless { flags } => self.test(&flags),
+            Subcommands::Run { flags } | Subcommands::RunDep { flags } => self.run_(&flags),
+            Subcommands::Fmt { flags } => self.fmt(&flags),
+            Subcommands::Clippy { flags } => self.clippy(&flags),
+            Subcommands::Cargo { flags } => self.cargo(&flags),
+            Subcommands::ManySeeds { command, seed_start, seeds } =>
+                self.many_seeds(&command, *seed_start, *seeds),
+            Subcommands::Bench { benches } => self.bench(&benches),
+            Subcommands::Toolchain { flags } => self.toolchain(&flags),
+            Subcommands::RustcPull { commit } => self.rustc_pull(commit.clone()),
+            Subcommands::RustcPush { github_user, branch } =>
+                self.rustc_push(&github_user, &branch),
+        }
+    }
+    fn find_sysroot(&mut self) {
+        let current_sysroot = std::env::var("MIRI_SYSROOT").unwrap_or_default();
+
+        if !current_sysroot.is_empty() {
```

Comment:
You are using `unwrap_or_default` a lot; this needs careful scrutiny.  Generally we should not just drop the `Option` like that.
For example here it should be
```suggestion
        let current_sysroot = std::env::var("MIRI_SYSROOT");

        if let Some(current_sysroot) = current_sysroot {
```

Reply:
I agree `unwrap_or_default` is a bit overused, but once again, this was a matter of parity with current Miri script. In some places it checks for variable's presence whereas in others it also checks whether it is not an empty string too.

Reply:
Yeah that's because it's really hard to do otherwise in a shell script.